_posts/2021-05-22-fluxv2-openfaas.md

```diff
@@ -0,0 +1,332 @@
+image: /images/2021-05-22-flux/earth.jpg
+categories:
+- arkade
+- kubectl
+- gitops
+- openfaas-operator
+- flux
+- CI/CD
+author_staff_member: alistair
+dark_background: true
+
+---
+
+We learn how to manage OpenFaaS and install functions using Flux.
+
+# What is Flux
+Flux is an open source project by [weaveworks](https://www.weave.works/oss/flux/) which enables 
+application delivery using GitOps. Your cluster config is stored in Git and the flux controllers continually work to ensure
+that the cluster config matches the state in Git. This allows us to take the "cattle not pets" mentality applied to modern cloud computing further and apply the same thinking
+to our clusters. 
+
+We no longer need to manually issue `helm install` or `kubectl apply` commands to manage our clusters and 
+configuration changes are audited and easily reviewed as Git PullRequests.
+
+![flux v2 diagram](/images/2021-05-22-flux/gitops-toolkit.png)
+*image from [Flux](https://github.com/fluxcd/flux2/blob/main/docs/_files/gitops-toolkit.png)*
+ 
+# GitOps
+
+`GitOps` is a set of practices that aim to democratise our "Operations" type practices such as deploying new application 
+versions or installing new software on servers that takes practices used in development such as using Git as a `single source of truth`
+and building auditability and accountability through peer review on Pull Requests to manage our day-to-day operations activities.
+
+I won't go into much mode detail as there are entire conferences dedicated to this topic and a few paragraphs from me won't 
+do this topic justice! Take some time to familiarise yourself with the concept if it is new to you. 
+
+### Assumptions
+
+* You have a Github Account (any git server can be used, please see section on installing flux for a link to the flux docs)
+
+
+## Tools
+We are going to use `arkade` to download our CLIs, you can install it like this
+
+```bash
+curl -SLs https://dl.get-arkade.dev | sh
+
+# For MacOS and Linux users:
+sudo mv arkade /usr/local/bin/
+
+# For Git Bash users:
+mv arkade /usr/bin/
+```
+
+Now get `kubectl`, `faas-cli`, `flux` and `kind` for use later.
+
+
+```bash
+arkade get kubectl
+arkade get kind
+arkade get faas-cli
+arkade get flux
+```
+
+follow the instructions in the output from the commands above to add the tools to your `$PATH`
+
+# Get a Kubernetes cluster
+
+Head on over to your favourite `Kubernetes as a Service` provider, we are using [linode](https://www.linode.com/) (An OpenFaaS Homepage sponsor)
+to provide us with our cluster. 
+
+You can also use `kind`, `k3d` or `minikube` to provision a cluster locally too!
+
+### Kubernetes on Linode
+1. Create an account on Linode, [this is the signup page](https://login.linode.com/signup). New users get $100 credit (at time of writing)
+
+2. Head to the `kubernetes` tab on the sidebar
+
+3. Create a cluster in a region close to you and select at least 3 nodes with 1cpu and 2GB ram. This should be enough for 
```

Comment:
Define "our applications"

I.e. This will be enough to deploy Flux, OpenFaaS and a few functions.